File: lihua-biz/lihua-system/src/main/java/com/lihua/entity/SysPost.java
package com.lihua.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.lihua.excel.annotation.ExcelDictType;
import com.lihua.excel.converter.ExcelDictConverter;
import com.lihua.mybatis.model.BaseEntity;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.apache.fesod.sheet.annotation.ExcelProperty;
import org.apache.fesod.sheet.annotation.write.style.ColumnWidth;

@EqualsAndHashCode(callSuper = true)
@Data
public class SysPost extends BaseEntity {

    /**
     * 主键
     */
    @TableId(type = IdType.ASSIGN_ID)
    private String id;

    /**
     * 部门主键
     */
    @NotNull(message = "请选择所属部门")
    private String deptId;

    /**
     * 部门编码
     */
    @NotNull(message = "请选择所属部门")
    private String deptCode;

    /**
     * 岗位名称
     */
    @NotNull(message = "请输入岗位名称")
    @ExcelProperty("岗位名称")
    @ColumnWidth(20)
    private String name;

    /**
     * 岗位编码
     */
    @NotNull(message = "请输入岗位编码")
    @ExcelProperty("岗位编码")
    @ColumnWidth(20)
    private String code;

    /**
     * 岗位排序
     */
    @NotNull(message = "请输入岗位排序")
    private Integer sort;

    /**
     * 岗位状态
     */
    @NotNull(message = "请选择状态")
    @ExcelProperty(value = "状态", converter = ExcelDictConverter.class)
    @ExcelDictType("sys_status")
    @ColumnWidth(10)
    private String status;

    /**
     * 岗位负责人姓名
     */
    @ExcelProperty("负责人")
    @ColumnWidth(20)
    private String manager;

    /**
     * 岗位联系电话
     */
    @Pattern(regexp = "^(|1[3-9]\\d{9})$",
            message = "请输入正确的手机号码")
    @ExcelProperty("联系电话")
    @ColumnWidth(20)
    private String phoneNumber;

    /**
     * 岗位邮件
     */
    @Pattern(regexp = "^(|[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,})$",
            message = "请输入正确的邮箱地址")
    @ExcelProperty("邮箱")
    @ColumnWidth(20)
    private String email;

    /**
     * 岗位传真号码
     */
    @ExcelProperty("传真")
    @ColumnWidth(20)
    private String fax;

    /**
     * 备注
     */
    @ExcelProperty("备注")
    @ColumnWidth(20)
    private String remark;

}

File: lihua-vue/src/api/system/post/type/sys-post.ts
export interface SysPost {
    /**
     * 主键
     */
    id?: string;

    /**
     * 部门主键
     */
    deptId?: string;

    /**
     * 部门编码
     */
    deptCode?: string;

    /**
     * 岗位名称
     */
    name?: string;

    /**
     * 岗位编码
     */
    code?: string;

    /**
     * 岗位排序
     */
    sort?: number;

    /**
     * 岗位状态
     */
    status?: string;

    /**
     * 岗位负责人姓名
     */
    manager?: string;

    /**
     * 岗位联系电话
     */
    phoneNumber?: string;

    /**
     * 岗位邮件
     */
    email?: string;

    /**
     * 岗位传真号码
     */
    fax?: string;

    /**
     * 备注
     */
    remark?: string;

    /**
     * 删除标识
     */
    delFlag?: string;
}

export interface SysPostVO extends SysPost{
    /**
     * 所属部门名称
     */
    deptName?: string

    statusIsNormal?: boolean

    updateStatusLoading?: boolean
}

export interface SysPostDTO {

    /**
     * 岗位状态
     */
    status?: string;

    /**
     * 岗位名称
     */
    name?: string;

    /**
     * 岗位编码
     */
    code?: string;

    /**
     * 所属部门id
     */
    deptId?: string | null;

    /**
     * 当前页数
     */
    pageNum: number;

    /**
     * 每页记录数
     */
    pageSize: number;
}